build/factory/memory-server-factory.js

```diff
@@ -0,0 +1,108 @@
+"use strict";
```

Comment:
usually we .gitignore build files, I'm curious why in this instance do we include them?

Reply:
For a GitHub Action, the code that will run when you 'use' it _must be javascript_ , so for an action written in TypeScript, the `action.yaml` file must point at some transpiled artifact (`build/index.js`, in this case).

* When a PR is created, the last step in that `pull-request` workflow is to diff the `./build` folder and do a commit as needed, providing said build artifact.

* If/When the PR is merged to master, the `publish` action will create a tag containing that pre-built artifact.